File: warehouse-management-web-angular/src/app/warehouse/document-shipment/models/output/resource-shipment-list-output.ts
import { DocumentStatusEnum } from '../../../enums/document-status-enum';
import { ResourceShipmentItemOutput } from './resource-shipment-item-output';

/**
 * Класс выходной модели ресурса документа отгрузки.
 */
export class ResourceShipmentListOutput {
  /**
   * Id документа отгрузки.
   */
  documentShipmentId: number = 0;

  /**
   * Номер документа отгрузки.
   */
  documentShipmentNumberCode: string = '';

  /**
   * Дата документа отгрузки.
   */
  documentShipmentDate: Date = new Date();

  /**
   * Id клиента документа отгрузки.
   */
  documentShipmentClientId: number = 0;

  /**
   * Наименование клиента документа отгрузки.
   */
  documentShipmentClientName: string = '';

  /**
   * Статус документа отгрузки в значении перечисления.
   */
  documentStatusEnum: DocumentStatusEnum = 0;

  /**
   * Список входящих ресурсов в документ отгрузки.
   */
  items: ResourceShipmentItemOutput[] = [];
}

File: WarehouseManagementWeb.Application/Dto/Output/ResourceShipment/ResourceShipmentListOutput.cs
using WarehouseManagementWeb.Domain.Enums;

namespace WarehouseManagementWeb.Application.Dto.Output.ResourceShipment
{
    /// <summary>
    /// Класс выходной модели ресурса документа отгрузки.
    /// </summary>
    public class ResourceShipmentListOutput
    {
        /// <summary>
        /// Id документа отгрузки.
        /// </summary>
        public int DocumentShipmentId { get; set; }

        /// <summary>
        /// Номер документа отгрузки.
        /// </summary>
        public string? DocumentShipmentNumberCode { get; set; }

        /// <summary>
        /// Дата документа отгрузки.
        /// </summary>
        public DateTime DocumentShipmentDate { get; set; }

        /// <summary>
        /// Id клиента документа отгрузки.
        /// </summary>
        public int DocumentShipmentClientId { get; set; }

        /// <summary>
        /// Наименование клиента документа отгрузки.
        /// </summary>
        public string? DocumentShipmentClientName { get; set; }

        /// <summary>
        /// Статус документа отгрузки в значении перечисления.
        /// </summary>
        public DocumentStatusEnum DocumentStatusEnum { get; set; }

        /// <summary>
        /// Список входящих ресурсов в документ отгрузки.
        /// </summary>
        public IEnumerable<ResourceShipmentItemOutput>? Items { get; set; }
    }
}
